utilization.md

```diff
@@ -1,142 +1,198 @@
+		1. [Create](#create)
+		2. [Manage](#manage)
+3. [Utilities](#utilities)
+	1. [Code interruption](#code-interruption)
+	2. [Print inside a kernel](#print-inside-a-kernel)
+	3. [Timer](#timer)
+		1. [Create](#create)
+		2. [Manage](#manage)
+	4. [Manage parallel environment](#manage-parallel-environment)
+4. [Macros](#macros)
+	1. [Essential macros](#essential-macros)
+	2. [Extra macros](#extra-macros)
 
 <!--#endif-->
 
-## Initialization
-
-### Headers
+## Header
 
 ```cpp
 #include <Kokkos_Core.hpp>
 ```
 
-### Initialize and finalize Kokkos
+## Initialization
+
+### Initialize and finalize
 
 ```cpp
-Kokkos::initialize(int& argc, char* argv[]);
-{
-    /* ... */
+int main(int argc, char* argv[]) {
+    Kokkos::initialize();
+    { /* ... */ }
+    Kokkos::finalize();
+    return 0;
 }
-Kokkos::finalize();
 ```
 
-#### Command-line arguments
+<!-- commented before this feature is tested -->
+<!-- ### Scope guard -->
+<!--  -->
+<!-- ```cpp -->
+<!-- int main(int argc, char* argv[]) { -->
+<!--     Kokkos::ScopeGuard kokkos(); -->
+<!--     /* ... */ -->
+<!--     return 0; -->
+<!-- } -->
+<!-- ``` -->
+
+## Kokkos concepts
 
-Command-line arguments are parsed by `Kokkos::initialize` and removed from the argument list.
+### Execution spaces
 
-| Argument                 | Effect                                                                                                                                              |
-|--------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `-kokkos-help`           | Print help message with the list of available options                                                                                               |
-| `-kokkos-threads=INT`    | Specify total number of threads or number of threads per NUMA region if used in conjunction with `--kokkos-numa` option                             |
-| `-kokkos-numa=INT`       | Specify number of NUMA regions used by process                                                                                                      |
-| `-device-id=INT`         | Specify device ID to be used by Kokkos                                                                                                              |
-| `-num-devices=INT[,INT]` | Specify the number of devices per node to be used when running MPI jobs; the optional second argument allows for an existing device to be ignored |
+| Execution space                     | Device backend | Host backend |
+|-------------------------------------|----------------|--------------|
+| `Kokkos::DefaultExecutionSpace`     | On device      | On host      |
+| `Kokkos::DefaultHostExecutionSpace` | On host        | On host      |
+| `Kokkos::Serial`                    | On host        | On host      |
```

Comment:
Why this backend?

Reply:
This was part of the original version. I'm not sure either why it's here, unless for explicit debug purpose.

Reply:
After discussion with Mathieu, we decided to remove this execution space.